hazelcast/src/test/java/com/hazelcast/scheduledexecutor/ScheduledExecutorSplitBrainTest.java

```diff
@@ -139,67 +160,80 @@ protected void onAfterSplitBrainHealed(HazelcastInstance[] instances) throws Exc
-        IScheduledExecutorService executorService1 = firstBrain[0].getScheduledExecutorService(name);
+    private void onAfterMergePassThroughMergePolicy() throws Exception {
+        assertContents(scheduledExecutorService1.<Double>getAllScheduledFutures());
+        assertContents(scheduledExecutorService2.<Double>getAllScheduledFutures());
+        assertHandlersAreStillCorrect();
+        assertUnexpectedFuturesHaveMergedValue();
+    }
 
+    private void onAfterSplitPutIfAbsentPolicy() {
+        // we should not see the tasks with UNEXPECTED_VALUE in the final tasks,
+        // since they have the same name as existing tasks
         for (int i = INITIAL_COUNT; i < AFTER_SPLIT_COMMON_COUNT; i++) {
-            schedule(executorService1, i);
+            schedule(scheduledExecutorService1, i, EXPECTED_VALUE);
+            schedule(scheduledExecutorService2, i, UNEXPECTED_VALUE);
         }
 
-        // Add up-to AFTER_SPLIT_COMMON_COUNT which should be ignored after merge, due to same name, and up-to FINAL_COUNT
-        // more tasks on top of that, which should be merged, unique names.
-        IScheduledExecutorService executorService2 = secondBrain[0].getScheduledExecutorService(name);
-        for (int i = INITIAL_COUNT; i < FINAL_COUNT; i++) {
-            schedule(executorService2, i);
+        // we should not lose these additional tasks, since they have a unique name
+        for (int i = AFTER_SPLIT_COMMON_COUNT; i < FINAL_COUNT; i++) {
+            schedule(scheduledExecutorService2, i, EXPECTED_VALUE);
         }
     }
 
-    private void onAfterMergePutAbsentMergePolicy(HazelcastInstance[] instances) throws Exception {
-        IScheduledExecutorService executorService = instances[0].getScheduledExecutorService(name);
-        assertContents(executorService.<Double>getAllScheduledFutures());
+    private void onAfterMergePutIfAbsentMergePolicy() throws Exception {
+        assertContents(scheduledExecutorService1.<Double>getAllScheduledFutures());
+        assertContents(scheduledExecutorService2.<Double>getAllScheduledFutures());
         assertHandlersAreStillCorrect();
+        assertUnexpectedFuturesHaveMergedValue();
     }
 
-    private void schedule(IScheduledExecutorService scheduledExecutorService, int value) {
-        // once a task runs, all calls to `.get()` should return 30
-        String name = valueOf(value);
-        IScheduledFuture future = scheduledExecutorService.schedule(
-                named(name, new ScheduledExecutorServiceTestSupport.PlainCallableTask(5)), 0, SECONDS);
-        allScheduledFutures.putIfAbsent(name, future);
+    private void schedule(IScheduledExecutorService scheduledExecutorService, int name, int taskValue) {
+        // once a task runs, all calls to future.get() should return PlainCallableTask.calculateResult(value)
+        String stringName = valueOf(name);
+        Callable<Double> task = named(stringName, new PlainCallableTask(taskValue));
```

Comment:
We can't just create all tasks with the same value, otherwise we can't check if an existing task really has been replaced. So we schedule tasks with `EXPECTED_VALUE` and `UNEXPECTED_VALUE` to have a distinguishable future result.